src/app/mytools/my-tool/my-tool.component.ts

```diff
@@ -116,22 +129,49 @@ export class MyToolComponent extends MyEntry implements OnInit {
     this.commonMyEntriesOnInit();
     this.containerService.setTool(null);
     this.containerService.setTools(null);
+    this.workflowService.setWorkflow(null);
+    this.workflowService.setWorkflows(null);
+    this.tool$ = this.toolQuery.tool$;
     this.toolQuery.tool$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tool) => {
       this.tool = tool;
     });
+    this.workflowQuery.workflow$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((appTool) => {
+      this.appTool = appTool;
+    });
 
     this.getMyEntries();
 
     this.containerService.tools$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tools) => {
       this.tools = tools;
-      this.selectEntry(this.mytoolsService.recomputeWhatEntryToSelect(tools));
+      if (this.apptools) {
+        this.allTools = this.apptools.concat(this.tools);
+      } else {
+        this.allTools = this.tools;
+      }
+      this.selectEntry(this.mytoolsService.recomputeWhatEntryToSelect(this.allTools));
     });
 
     this.groupEntriesObject$ = combineLatest([this.containerService.tools$, this.toolQuery.tool$]).pipe(
       map(([tools, tool]) => {
         return this.mytoolsService.convertEntriesToOrgEntryObject(tools, tool);
       })
     );
+
+    this.workflowService.workflows$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((appTools) => {
+      this.apptools = appTools;
+      if (this.tools) {
+        this.allTools = this.apptools.concat(this.tools);
```

Comment:
this.allTools is also set above in the containerService.tools$ subscribe.  forkjoin/combinelatest is more suitable